TUnit.Assertions.SourceGenerator/Helpers/AttributeExtractors/GenerateAssertionDto.cs

```diff
@@ -0,0 +1,235 @@
+﻿// ---------------------------------------------------------------------------------------------------------------------
+// Imports
+// ---------------------------------------------------------------------------------------------------------------------
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace TUnit.Assertions.SourceGenerator.Helpers.AttributeExtractors;
+
+// ---------------------------------------------------------------------------------------------------------------------
+// Code
+// ---------------------------------------------------------------------------------------------------------------------
+public class GenerateAssertionDto(
+    Location? attributeLocation,
+    ITypeSymbol typeArg,
+    AssertionType assertionType,
+    string methodName,
+    string? messageFactoryMethodName,
+    string? expectationExpression
+) {
+    public bool RequiresNullCheck => !typeArg.IsValueType;
+    private readonly HashSet<char> _vowels = ['a', 'e', 'i', 'o', 'u'];
+    private ISymbol? _memberSymbol;
+
+    public string GetTypeName() {
+        string typeName = typeArg.Name;
+        if (typeArg.SpecialType is
+            SpecialType.System_Char
+            or SpecialType.System_String
+            or SpecialType.System_Boolean
+            or SpecialType.System_Int32
+            or SpecialType.System_Double
+            or SpecialType.System_Single
+            or SpecialType.System_Byte) {
+            typeName = typeName.ToLowerInvariant();
+        }
+
+        return typeName;
+    }
+
+    public string GetMethodName() {
+        // ReSharper disable once SwitchExpressionHandlesSomeKnownEnumValuesWithExceptionInDefault
+        return assertionType switch {
+            AssertionType.Is => methodName,
+            AssertionType.IsNot => methodName.Replace("Is", "IsNot"),
+            _ => throw new ArgumentOutOfRangeException(nameof(assertionType), assertionType, null),
+        };
+    }
+
+    public string GetMessageFactoryOrDefault() {
+        if (!string.IsNullOrWhiteSpace(messageFactoryMethodName)) return messageFactoryMethodName!;
+
+        string strippedMethodName = methodName.Replace("Is", "");
+        string a = _vowels.Contains(strippedMethodName.ToLowerInvariant()[0]) ? "an" : "a";
+
+        // ReSharper disable once SwitchExpressionHandlesSomeKnownEnumValuesWithExceptionInDefault
+        return assertionType switch {
+            AssertionType.Is => $"(s, _, _) => \"'{{s}}' was not {a} {strippedMethodName}\"",
```

Comment:
We want string interpolation here I think?
```suggestion
            AssertionType.Is => $"(s, _, _) => $\"'{{s}}' was not {a} {strippedMethodName}\"",
```